css/src/components/checkbox.scss

```diff
@@ -0,0 +1,118 @@
+/* stylelint-disable selector-max-specificity, max-nesting-depth, selector-no-qualifying-type */
+
+$checkbox-border-radius: $radius-sm !default;
+$checkbox-border-color: $text !default;
+$checkbox-check-hover-color: $secondary !default;
+$checkbox-icon-size: 0.875rem !default;
+$checkbox-size: 1.25rem !default;
+$checkbox-size-small: 0.875rem !default;
+$checkbox-icon-size: 1rem !default;
+$checkbox-icon-size-small: 0.625rem !default;
+$checkbox-timing-function: $input-timing-function !default;
+$checkbox-duration: $input-transition-duration !default;
+$checkbox-spacing: $spacer-3;
+
+.checkbox {
+	display: flex;
+	position: relative;
+	align-items: center;
+	line-height: 1.25;
+	cursor: pointer;
+
+	+ .checkbox {
+		margin-top: $checkbox-spacing;
+		margin-left: 0;
+	}
+
+	.checkbox-check {
+		display: flex;
+		flex-shrink: 0;
+		align-items: center;
+		justify-content: center;
+		width: $checkbox-size;
+		height: $checkbox-size;
+		transition-duration: $checkbox-duration;
+		transition-property: background, border, border-color;
+		transition-timing-function: $checkbox-timing-function;
+		border: 1px solid $checkbox-border-color;
```

Comment:
Think that our border needs some work, since it needs to be the same color as the background when its checked:

This PR: 
![image](https://user-images.githubusercontent.com/30843002/148825427-3e8c0936-7974-4d08-8b28-08298de4ac7a.png)

Figma
![image](https://user-images.githubusercontent.com/30843002/148825445-74359c0e-e67d-4940-a249-db5ab45601b9.png)

Old:
![image](https://user-images.githubusercontent.com/30843002/148825571-6a1d8d62-2fd4-48c8-958c-e02d2d793434.png)

Reply:
Updated border color. I made a slight adjustment to the checkmark size, but it may require more tweaking.